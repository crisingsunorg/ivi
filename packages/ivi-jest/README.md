# Jest utils for ivi library
