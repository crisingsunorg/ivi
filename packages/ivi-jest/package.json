{
  "private": true,
  "name": "ivi-jest",
  "version": "0.0.1",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "sideEffects": false,
  "files": [
    "dist",
    "src",
    "!dist/**/*.tsbuildinfo",
    "!src/**/__tests__",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "prepublishOnly": "npm run clean && npm run dist",
    "clean": "rm -rf ./dist",
    "dist": "npm run dist:main && npm run dist:module",
    "dist:main": "../../node_modules/.bin/tsc -p ./tsconfig.build.main.json",
    "dist:module": "../../node_modules/.bin/tsc -p ./tsconfig.build.module.json"
  },
  "dependencies": {
    "jsdom": "^14.0.0"
  },
  "peerDependencies": {},
  "devDependencies": {},
  "description": "Jest utils for ivi library",
  "license": "MIT",
  "author": {
    "name": "Boris Kaul",
    "email": "localvoid@gmail.com",
    "url": "https://github.com/localvoid"
  },
  "keywords": [
    "ivi"
  ],
  "homepage": "https://github.com/localvoid/ivi",
  "repository": {
    "type": "git",
    "url": "https://github.com/localvoid/ivi"
  },
  "bugs": {
    "url": "https://github.com/localvoid/ivi/issues"
  },
  "github": "https://github.com/localvoid/ivi"
}
