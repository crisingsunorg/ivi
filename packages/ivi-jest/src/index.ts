export { useMockFn, useSpyOn } from "./mock";
export { useResetModules, useModule } from "./module";
export { useDOMElement, useDOMOpsCounters } from "./dom";
