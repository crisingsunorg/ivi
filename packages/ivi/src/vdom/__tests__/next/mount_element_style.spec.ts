import { useResetModules, useModule, useDOMElement } from "ivi-jest";

useResetModules();
const c = useDOMElement();
const h = useModule<typeof import("ivi-html")>("ivi-html");
const t = useModule<typeof import("ivi-test")>("ivi-test");
const _ = void 0;

describe("mount element style", () => {
  test("undefined", () => {
    const { domNode } = t.render(h.div(_, { style: void 0 }), c());
    expect(domNode).toMatchSnapshot();
  });

  test("property with undefined value", () => {
    const { domNode } = t.render(h.div(_, { style: { top: void 0 } }), c());
    expect(domNode).toMatchSnapshot();
  });

  test("one property", () => {
    const { domNode } = t.render(h.div(_, { style: { top: "10px" } }), c());
    expect(domNode).toMatchSnapshot();
  });

  test("two properties", () => {
    const { domNode } = t.render(h.div(_, { style: { top: "10px", left: "20px" } }), c());
    expect(domNode).toMatchSnapshot();
  });
});
