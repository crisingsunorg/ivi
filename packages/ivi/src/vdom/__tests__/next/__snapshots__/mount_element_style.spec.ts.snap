// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount element style one property 1`] = `
<div
  style="top: 10px;"
/>
`;

exports[`mount element style property with undefined value 1`] = `<div />`;

exports[`mount element style two properties 1`] = `
<div
  style="top: 10px; left: 20px;"
/>
`;

exports[`mount element style undefined 1`] = `<div />`;
